File: leetcode/1200. Minimum Absolute Difference/1200.cpp
// https://helloacm.com/the-minimum-absolute-difference-algorithm-of-an-array/
// https://leetcode.com/problems/minimum-absolute-difference/

class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> r;
        sort(begin(arr), end(arr));        
        int D = INT_MAX;
        for (int i = 1; i < arr.size(); ++ i) {
            int d = arr[i] - arr[i - 1];
            if (d < D) {
                r.clear();
                D = d;
                r.push_back({arr[i - 1], arr[i]});
            } else if (d == D) {
                r.push_back({arr[i - 1], arr[i]});
            }
        }        
        return r;
    }
};

File: leetcode/1200. Minimum Absolute Difference/1200.py
# https://helloacm.com/teaching-kids-programming-compute-minimum-absolute-difference-of-two-numbers-in-an-array/
# https://leetcode.com/problems/minimum-absolute-difference/
# EASY, SORTING

class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        n = len(arr)
        arr.sort()
        minDiff = math.inf
        ans = []
        for i in range(1, n):
            curDiff = arr[i] - arr[i - 1]
            if curDiff < minDiff:
                minDiff = curDiff
                ans = [[arr[i - 1], arr[i]]]
            elif curDiff == minDiff:
                ans.append([arr[i - 1], arr[i]])
        return ans
